src/components/timeline/GlobalTrack.js

```diff
@@ -0,0 +1,279 @@
+        </div>
+        <Reorderable
+          tagName="ol"
+          className="timelineTrackLocalTracks"
+          order={localTrackOrder}
+          orient="vertical"
+          grippyClassName="timelineTrackLocalGrippy"
+          onChangeOrder={this._changeLocalTrackOrder}
+        >
+          {localTracks.map((localTrack, trackIndex) => {
+            if (pid === null) {
+              console.error(
+                'The pid should never be null when adding a TimelineLocalTrack.',
+                pid
+              );
+              return <div />;
+            }
+            return (
+              <TimelineLocalTrack
+                key={trackIndex}
+                pid={pid}
+                localTrack={localTrack}
+                trackIndex={trackIndex}
+              />
+            );
+          })}
+        </Reorderable>
+      </li>
+    );
+  }
+}
+
+function getGlobalTrackName(
+  globalTrack: GlobalTrack,
+  threads: Thread[]
+): string {
+  switch (globalTrack.type) {
+    case 'process': {
+      // Look up the thread information for the process if it exists.
+      return globalTrack.mainThreadIndex === null
+        ? `Process ${globalTrack.pid}`
+        : getFriendlyThreadName(threads, threads[globalTrack.mainThreadIndex]);
+    }
+    case 'screenshots':
+      return 'Screenshots';
+    default:
+      throw new Error(`Unhandled GlobalTrack type ${(globalTrack: empty)}`);
+  }
+}
+
+// Provide some empty lists, so that strict equality checks work for component updates.
+const EMPTY_TRACK_ORDER = [];
+const EMPTY_LOCAL_TRACKS = [];
+
+const options: ExplicitConnectOptions<OwnProps, StateProps, DispatchProps> = {
+  mapStateToProps: (state, { trackIndex }) => {
+    const globalTracks = getGlobalTracks(state);
+    const globalTrack = globalTracks[trackIndex];
+
+    // These get assigned based on the track type.
+    let threadIndex = null;
+    let isSelected = false;
+    let titleText = null;
+    let trackName = getGlobalTrackName(globalTrack, getThreads(state));
+
+    let localTrackOrder = EMPTY_TRACK_ORDER;
+    let localTracks = EMPTY_LOCAL_TRACKS;
+    let pid = null;
+
+    // Run different selectors based on the track type.
+    switch (globalTrack.type) {
+      case 'process':
+        {
+          // Look up the thread information for the process if it exists.
+          if (globalTrack.mainThreadIndex !== null) {
+            threadIndex = globalTrack.mainThreadIndex;
+            const selectors = selectorsForThread(threadIndex);
+            isSelected = threadIndex === getSelectedThreadIndex(state);
+            trackName = selectors.getFriendlyThreadName(state);
+            titleText = selectors.getThreadProcessDetails(state);
+          } else {
+            trackName = `Process ${globalTrack.pid}`;
+          }
+          pid = globalTrack.pid;
+          localTrackOrder = getLocalTrackOrder(state, pid);
+          localTracks = getLocalTracks(state, pid);
+        }
+        break;
```

Comment:
nit: you can put `break` inside the brackets above to get a bit better indentation